src/main/java/Duke.java

```diff
@@ -1,10 +1,251 @@
+     */
+    public static boolean isEmptyItem(String args) {
+        if (getItem(args).equals("")) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * returns if there is a time entered for a deadline or event task
+     *
+     * @param args the input string the user entered
+     * @return true if there is a time entered, false otherwise
+     */
+    public static boolean canGetTime(String args) {
+        String description = getItem(args);
+        String[] time = description.substring(description.indexOf("/")).split(" ");
+        if (time.length < 2) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Checks if the item from the user input is invalid for deadline and event tasks
+     *
+     * @param args the item to be checked
+     * @return true if the item does not contain a '/by' or does not have time after /by, false otherwise
+     */
+    public static boolean isInvalidDeadline(String args) {
+        if (!getItem(args).contains(BY_DIVIDER)) {
+            return true;
+        }
+        if (!canGetTime(args)) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Checks if the item from the user input is invalid for deadline and event tasks
+     *
+     * @param args the item to be checked
+     * @return true if the item does not contain a '/at' or does not have time after /at, false otherwise
+     */
+    public static boolean isInvalidEvent(String args) {
+        if (!getItem(args).contains(AT_DIVIDER)) {
+            return true;
+        }
+        if (!canGetTime(args)) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Reads the input and command entered by the user and execute task according to user command
+     *
+     * @param inputStr the full input the user entered
+     * @param command the command parsed from the input
+     * @param taskManager the task manager instance handling the tasks
+     * @param fileManager the file manager handling the files
+     *
+     **/
+    public static void executeCommand(String inputStr, String command, TaskManager taskManager, FileManager fileManager) throws DukeException {
+        switch (command) {
+        case COMMAND_EXIT:
+            fileManager.writeArrayToFile(taskManager);
+            System.out.println(LINE_SEPARATOR);
+            System.out.println("Bye. Hope to see you again soon!");
+            System.out.println(LINE_SEPARATOR);
+            canRunDuke = false;
+            break;
+        case COMMAND_LIST:
+            taskManager.printTaskList();
+            break;
+        case COMMAND_DONE:
+            if (isEmptyItem(inputStr)) {
+                throw new DukeException("Oops, did you forget to enter the task to be marked as done?");
+            } else if (taskManager.getNumberOfTasksUndone() == 0) {
+                throw new DukeException("Oops, there are no tasks to be marked done!");
```

Comment:
Even error messages are considered as string constants/literals so you should add them as named constants.